docs/index.rst

```diff
@@ -5,13 +5,98 @@
 
 Welcome to nanopub's documentation!
 ===================================
-Documentation for nanopub is still work in progress
+The ``nanopub`` library provides a high-level, user-friendly python
+interface for searching, publishing and retracting nanopublications.
+Nanopublications are FAIR data containers for scientific results
+and more, read more about them on http://nanopub.org/.
 
-.. toctree::
-   :maxdepth: 2
-   :caption: Contents:
+Setup
+-----
 
+Install using pip:
 
+::
+
+    pip install nanopub
+
+To publish to the nanopub server you need to setup your profile. This
+allows the nanopub server to identify you. Run the following interactive
+command:
+
+::
+
+    setup_profile
+
+It will add and store RSA keys to sign your nanopublications, publish a
+nanopublication with your name and ORCID iD to declare that you are
+using using these RSA keys, and store your ORCID iD to automatically add
+as author to the provenance of any nanopublication you will publish
+using this library.
+
+Quick Start
+-----------
+
+Publishing nanopublications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: python
+
+    import rdflib
+    from nanopub import Publication, NanopubClient
+
+    # Create the client (we use use_test_server=True to point to the test server)
+    client = NanopubClient(use_test_server=True)
+
+    # Either quickly publish a statement to the server
+    client.claim('All cats are gray')
+
+    # Or: 1. construct a desired assertion (a graph of RDF triples) using rdflib
+    my_assertion = rdflib.Graph()
+    my_assertion.add((rdflib.URIRef('www.example.org/timbernerslee'),
+                      rdflib.RDF.type,
+                      rdflib.FOAF.Person))
+
+    # 2. Make a Publication object with this assertion
+    publication = Publication.from_assertion(assertion_rdf=my_assertion)
+
+    # 3. Publish the Publication object. The URI at which it is published is returned.
+    publication_info = client.publish(publication)
+    print(publication_info)
+
+Searching for nanopublications
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code:: python
+
+    from nanopub import NanopubClient
+
+    # Create the client
+    client = NanopubClient()
+
+    # Search for all nanopublications containing the text 'fair'
+    results = client.find_nanopubs_with_text('fair')
+    print(results)
+
```

Comment:
Do we want to indicate here, even just with a code comment, that there are other search methods other than plain text?

Reply:
Yeah we should add a link to the more detailed section on searching!

Reply:
I've updated #37 so we don't forget.